### DISCUSSION TITLE
Amplify SSR - how to view logs and debug?

### DISCUSSION THREAD

**Discussion by DiegoMcDipster** [vercel/next.js#33457] [Help]
Hi,

I was getting errors on my NextJs Amplify app while fetching data in getServerSideProps. I found the root of the problem by... I'm a bit embarassed to say... writing the error message to cookies :s

But surely there's a better way? My question is what's the best way of debugging errors serverside on AWS Amplify? For example, the lambda function logging can be seen in cloudwatch, but my error was occuring before executing the lambda.  So... if i'd written console.logs on my GSSP, how could they have been viewed? ... in cloudwatch?

**Comment by socketbox** [comment]
Did you have a look at this? https://github.com/aws-amplify/amplify-hosting/blob/main/FAQ.md#access-lambda-edge-logs

**Reply by KthProg** [reply]
I think this may be outdated. When I go to deployments, there is no lambda mentioned in the build logs. I also went to CloudWatch and confirmed there are no new log groups starting with /aws/lambda.
